.github/workflows/release.yml

```diff
@@ -0,0 +1,59 @@
+on:
+  push:
+    # Sequence of patterns matched against refs/tags
+    tags:
+    - 'v*' # Push events to matching v*, i.e. v1.0, v20.15.10
+name: Handle Release
+jobs:
+  build:
+    name: Upload Release Asset
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@master
+        
+      - name: Set GOPATH
+      # temporary fix
+      # see https://github.com/actions/setup-go/issues/14
+        run: |
+          echo "##[set-env name=GOPATH;]$(dirname $GITHUB_WORKSPACE)"
+          echo "##[add-path]$(dirname $GITHUB_WORKSPACE)/bin"
+        shell: bash
+    
+      - name: Prepare environment
+        run: |
+          go mod download
+          hack/install-gox.sh\
+
+      - name: Install dependencies
+        run: |
+          go mod tidy && git diff --no-patch --exit-code
+          hack/verify-code-patterns.sh
+          hack/verify-boilerplate.sh
+          hack/run-lint.sh
+          go test -short -coverprofile=coverage.txt -covermode=atomic ./...
+          hack/make-all.sh
+          hack/verify-installation.sh
+          hack/ensure-kubectl-installed.sh
+          hack/run-integration-tests.sh
+          hack/verify-receipts-upgrade-migration.sh
+
+      - name: Create Release
+        id: create_release
+        uses: actions/create-release@v1.0.0
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          tag_name: ${{ github.ref }}
+          release_name: Release ${{ github.ref }}
+          draft: false
+          prerelease: false
+      
+      - name: Upload binaries to release
+        uses: svenstaro/upload-release-action@v1-release
+        with:
+          repo_token: ${{ secrets.GITHUB_TOKEN }}
+          file: out/krew.*
+          tag: ${{ github.ref }}
+          overwrite: true
+          file_glob: true
```

Comment:
it seems like this is the only relevant part and the first ~40 lines are duplicated from ci.yml.

is there a way to de-duplicate?

Reply:
The problem I found is GitHub actions does not have option to ejecute multiples Jobs, I mean I did not found a way to join build stage and deploy stage in the same file because, it runs in different ways(push, PR, tag), so the solution by GitHub comunity is run it in different files. That's why I had to duplicate that part, to deploy artifacts.

Reply:
if trigger is extended, then `if conditional` should work but not familiar with the exact syntax. 

A suggestion:

`if: github.event_name == 'push' && github.ref == 'v\d+\.\d+.\d+.*'`

Reply:
I ran to this problem earlier as well

https://github.com/ahmetb/kubectl-tree/blob/master/.github/workflows/ci.yml
https://github.com/ahmetb/kubectl-tree/blob/master/.github/workflows/release.yml

and I don't like the duplication :( Let's keep searching?

Reply:
Sure, I'll keep searching for a while.
@ahmetb Anyway, this still needs much to improve but my suggestion is to move forward in parallel with the improvements of GitHub, I could follow this process for a long time. And see if we can improve the other Kubernetes projects with GitHub actions.

Reply:
Yes, I did. You can check it now :)